src/translator/scorers.cpp

```diff
@@ -91,11 +89,15 @@ std::vector<Ptr<Scorer>> createScorers(Ptr<Config> options,
   size_t i = 0;
   for(auto ptr : ptrs) {
     std::string fname = "F" + std::to_string(i);
-    auto modelOptions = New<Config>(*options);
 
+    // load options specific for the scorer
+    auto modelOptions = New<Options>(*options);
     try {
-      if(!options->get<bool>("ignore-model-config"))
-        modelOptions->loadModelParameters(ptr);
+      if(!options->get<bool>("ignore-model-config")) {
+        YAML::Node modelYaml;
+        io::getYamlFromModel(modelYaml, "special:model.yml", ptr);
+        modelOptions->merge(modelYaml, true);
+      }
```

Comment:
This should be hidden. Maybe a static function in Config:: ? It is modifying an options object, but also clearly a configuration-specific operation.

Reply:
Agree. I aimed for minimal changes.